nativelib/src/main/scala-3/scala/scalanative/unsafe/UnsafePackageCompat.scala

```diff
@@ -26,27 +26,48 @@ private[scalanative] trait UnsafePackageCompat {
   inline def ssizeof[T]: CSSize = fromRawSize(Intrinsics.sizeOf[T])
 
   /** Heap allocate and zero-initialize value using current implicit allocator.
+   *
+   *  Alias for preferred 'calloc*()' equivalent. Issue #4700
    */
-  inline def alloc[T]()(using zone: Zone): Ptr[T] = {
-    val size = sizeof[T]
-    val ptr = zone.alloc(size)
-    ffi.memset(ptr, 0, size)
-    ptr.asInstanceOf[Ptr[T]]
-  }
+  inline def alloc[T]()(using zone: Zone): Ptr[T] =
+    calloc[T]()
 
   /** Heap allocate and zero-initialize n values using current implicit
    *  allocator.
+   *
+   *  Alias for preferred 'calloc*()' equivalent.
    */
   inline def alloc[T](inline n: CSize)(using zone: Zone): Ptr[T] =
-    alloc[T](toRawSize(n))
+    calloc[T](toRawSize(n))
 
   /** Heap allocate and zero-initialize n values using current implicit
    *  allocator.
+   *
+   *  Alias for preferred 'calloc*()' equivalent.
    */
   inline def alloc[T](inline n: Int)(using zone: Zone): Ptr[T] =
-    alloc[T](validateSize(n))
+    calloc[T](validateSize(n))
+
+  /** Heap allocate and zero-initialize value using current implicit allocator.
+   *
+   *  Prefer 'calloc*()' over 'alloc*()'equivalent. Issue #4700
+   */
+  inline def calloc[T]()(using zone: Zone): Ptr[T] =
+    calloc[T](sizeof[T])
```

Comment:
i think it should be `(1)`, or you'll allocate `sizeof[T] * sizeof[T]`... (i wonder how to unit test the allocated sizes).

two other alternatives:
- keep the old remove `alloc` on red line 30, just rename it as calloc; or
- change the one on green line 67 to be `inline n: Int = 1`